### PR TITLE
Symmetric solid angle calculation for WcsGeom

### PR DESCRIPTION
**Pull request by watsonjj**
Currently the solid angle calculation for a `WcsGeom` is not symmetric about the origin. This is a result of calculating the separation along the bin edge instead of along the bin centre.

### CODE REVIEW

**Comment by watsonjj** [comment]
I will explain the problem, and what is causing it:

image

In the image above I have created a Wcs Map centred on the origin with three bins in each axis. Currently the dx values for the solid angle calculation are calculated along the bottom of the pixels as shown in the above figure. Due to the great circle separations of the coordinate frame, the separation between the lon coordinates of the pixel edges reduces as you increase/decrease in lat. As the bottom of the pixels is symmetric for the top two pixels, they have the same x separation. They will therefore result in the same solid angle. Whereas the bottom pixel will have a different solid angle value, as x2 is measured at a further distance along lat from the origin.

The problem I see with this is the solid angle measured for each pixel is not symmetric about the origin. I would have expected the top and bottom pixel to have the same solid angle value, as they both have the same distance along the lat axis.

image

If one instead measures the separation along the centre of the pixel, as show above, then the calculation is symmetric about the origin, and the solid angles come out as expected.

image

The above image shows the expected area distribution across pixels.

File: gammapy/maps/wcs.py [hunk 1]
```diff
@@ -773,15 +773,20 @@ def solid_angle(self):
...
 
-        # Compute solid angle using the approximation that it's
-        # the product between angular separation of pixel corners.
+        # Compute solid angle across centres of the pixels, approximating it
+        # as a rectangle
         # First index is "y", second index is "x"
-        ylo_xlo = lon[..., :-1, :-1], lat[..., :-1, :-1]
-        ylo_xhi = lon[..., :-1, 1:], lat[..., :-1, 1:]
-        yhi_xlo = lon[..., 1:, :-1], lat[..., 1:, :-1]
+        # TODO: Calculate actual solid angle between two great circles?
```
Comment (registerrier) [inline gammapy/maps/wcs.py]: A priori I think that one can use great circle only for very specific `WCS` projections, e.g. for the Plate-Carrée or `CAR` projection. But for most other projections the axes do not follow great circles e.g. Hammer-Aitoff `AIT` or tangential `TAN`.

So it is possible to use exact formulae for some specific projections, but not in general.

**Comment by watsonjj** [comment]
> How significant is the change in the solid angle results for typical test cases we have implemented. I imagine there should be some change in many gammapy.cube tests, no?

I have committed the changes required for other tests that had a small (on the order of 0.01%) change in the fixed values

File: gammapy/maps/wcs.py (hunk 1 above)
Comment (watsonjj) [reply gammapy/maps/wcs.py]: Okay. Shall I remove this TODO?

**Pull request merged**